### ISSUE TITLE
False positive reportInvalidTypeVarUse when using multiple class generic types

### ISSUE THREAD

**Issue by zakstucke** [microsoft/pyright#6402]
Moving from mypy, when using multiple types for a generic class and expanding prior types to access a later one pyright complains about unused types. 

I've searched around for someone hitting the same but can't quite workout whether this is a bug, or there's some easy native fix to access a specific type var and avoid this, sorry if the latter!

```python
import typing as tp

A = tp.TypeVar("A")
B = tp.TypeVar("B")


class Foo(tp.Generic[A, B]):
    a: A
    b: B


def use_foo(foo: Foo[A, B]) -> B: # TypeVar "A" appears only once in generic function signature (reportInvalidTypeVarUse)
    return foo.b
```

Pyright: 1.1.335
Python: 3.11

**Comment by erictraut** [comment]
In your example, TypeVar `A` is meaningless and can simply be replaced by `Any`. It's not technically a "bug" to include it, but its presence indicates that the user may not have an understanding of how TypeVars work and may be using them in an incorrect manner. I added the `reportInvalidTypeVarUse` check after seeing many Python programmers stumble with their use.

PEP 695 (which I authored — and is implemented in Python 3.12) makes this clearer.

```python
from typing import Any

class Foo[A, B]:
    a: A
    b: B

def use_foo[B](foo: Foo[Any, B]) -> B:
    return foo.b
```

**Comment by zakstucke** [comment]
Thankyou @erictraut, it's so simple once you see it, sorry for wasting your time!

Can't wait to try out 3.12 with how much simpler its making type variables, thanks for all your hard work!
